frontend/src/lib/query.js

```diff
@@ -390,6 +393,40 @@ var Query = {
         }
     },
 
+    addExpression(query) {
+        let expressions = query.expressions || {};
+        expressions[''] = [null, ''];
+        query.expressions = expressions;
+
+        console.log('added expressions. expressions =', query.expressions);
+    },
+
+    updateExpression(query, name, expression) {
+        query.expressions[name] = expression;
+
+        console.log('updated expression. expressions =', query.expressions);
+    },
+
+    renameExpression(query, oldName, newName) {
+        if (oldName === newName) return;
+
+        oldName = oldName || '';
+        query.expressions[newName] = query.expressions[oldName];
+        delete query.expressions[oldName];
+
+        console.log('renamed expression. expressions =', query.expressions);
+    },
+
+    removeExpression(query, name) {
+        if (!query.expressions) return;
+
+        delete query.expressions[name];
+
+        if (_.isEmpty(query.expressions)) delete query.expressions;
+
+        console.log('removed expression. expressions =', query.expressions);
+    },
```

Comment:
for these expression clause specific function lets create a separate import that can be used independently.  look further down in the query.js file for an example, e.g. `AggregationClause`